topics/admin/tutorials/ansible-galaxy/tutorial.md

```diff
@@ -2073,6 +2228,101 @@ Firstly, the plugins section contains a plugin called "local" which is of type "
 >
 {: .hands_on}
 
+## Productionising Galaxy
+
+This is a fantastic base Galaxy installation but there are numerous additional options we should be setting for a real production Galaxy.
+
+> ### {% icon hands_on %} Hands-on: More Features!
+>
+> 1. Make the following changes to your `group_vars/galaxyservers.yml`:
+>
+>    {% raw %}
+>    ```diff
+>    --- a/group_vars/galaxyservers.yml
+>    +++ b/group_vars/galaxyservers.yml
+>    @@ -38,6 +38,28 @@ galaxy_config:
+>         object_store_store_by: uuid
+>         id_secret: "{{ vault_id_secret }}"
+>         job_config_file: "{{ galaxy_config_dir }}/job_conf.xml"
+>    +    # SQL Performance
+>    +    database_engine_option_server_side_cursors: true
+>    +    slow_query_log_threshold: 5
+>    +    enable_per_request_sql_debugging: true
+>    +    # File serving Performance
+>    +    nginx_x_accel_redirect_base: /_x_accel_redirect
+>    +    # Automation / Ease of Use / User-facing features
+>    +    watch_job_rules: 'auto'
+>    +    allow_path_paste: true
+>    +    enable_quotas: true
+>    +    allow_user_deletion: true
+>    +    show_welcome_with_login: true
+>    +    expose_user_name: true
+>    +    expose_dataset_path: true
+>    +    expose_potentially_sensitive_job_metrics: true
+>    +    # NFS workarounds
+>    +    retry_job_output_collection: 3
+>    +    # Debugging
+>    +    cleanup_job: onerror
+>    +    allow_user_impersonation: true
+>    +    # Tool security
+>    +    outputs_to_working_directory: true
+>       uwsgi:
+>         socket: 127.0.0.1:8080
+>         buffer-size: 16384
+>    {% endraw %}
+>    ```
+>    {: data-commit="Add production facing vars"}
+>
+>    > ### {% icon tip %} What do these do?
+>    > Check out the full details in the [Galaxy documentation](https://docs.galaxyproject.org/en/{{ galaxy_version }}/admin/config.html#configuration-options), but we'll discuss a couple briefly:
+>    > - `nginx_x_accel_redirect_base`: This is required to have NGINX serve user files. You don't want Galaxy to waste time reading a 100GB fastq file a user has asked for, so you offload that to NGINX. The request is passed through to Galaxy, so permissions checks still occur, but Galaxy instead replies to NGINX just the path to the file that it should send to the requesting user.
+>    > - `enable_quotas`: You definitely want to set a default quota for your users!
+>    > - `expose_user_name`: This exposes usernames in the history and dataset sharing forms which makes life easier for your users.
+>    > - `expose_dataset_path`: This shares the entire dataset path.
```

Comment:
I would add that this also enable the user to have access to the command line that the tool launched.

Reply:
If you'd like to use the suggestion feature to suggest a wording change, I could include that!